device-connectors/src/testflinger_device_connectors/devices/zapper_kvm/__init__.py

```diff
@@ -116,31 +116,43 @@ def _validate_configuration(
         """Validate the job config and data and prepare the arguments
         for the Zapper `provision` API.
         """
-        if "alloem_url" in self.job_data["provision_data"]:
-            url = self.job_data["provision_data"]["alloem_url"]
-            username = "ubuntu"
-            password = "u"
-            retries = max(
+        provisioning_data = {}
+        if "preset" in self.job_data["provision_data"]:
+            for key, value in self.job_data["provision_data"].items():
+                provisioning_data[key] = value
+
```

Comment:
`username`/`password` need special attention: they are part of the test phase, not provisioning. we probably want to *always* override them, and you should create a dedicated function to return them based on the other inputs. Each case in this if/else statement should make use of this new function. As you can see below, if `alloem` is provided, username/password are special. Otherwise, they're taken from test_data with default values.